libsolidity/ast/Types.cpp

```diff
@@ -949,11 +949,22 @@ bool RationalNumberType::operator==(Type const& _other) const
 	return m_value == other.m_value;
 }
 
+string numberToString(dev::bigint num)
+{
+	string str = num.str();
+	if (str.size() > 32)
+	{
+		int omitted = str.size() - 8;
+		return str.substr(0, 4) + "...(" + to_string(omitted) + " digits omitted)..." + str.substr(str.size() - 4, 4);
+	}
+	return str;
+}
+
 string RationalNumberType::toString(bool) const
```

Comment:
Are you sure `toString` is not used anywhere else but error messages? I think it may be used in `ASTJsonConverter`.

Reply:
You are right, I missed those usages. Are these type strings documented somewhere? Perhaps we can take a moment and describe them in more detail.

Reply:
the bool parameter indicates whether it is for human or machine consumption, I think.

Reply:
For some use-cases it is important that if two types compare unequal, the `toString` output has to be different.

Reply:
bool parameters are THE worst :grimacing:

Reply:
Based on this `true` stands for machine use?

```
        virtual std::string toString(bool _short) const = 0;
        std::string toString() const { return toString(false); }
        /// @returns the canonical name of this type for use in library function signatures.
        virtual std::string canonicalName() const { return toString(true); }
```

Reply:
The documentation is in the base class.

Reply:
This is from the base class.